overlay/files/system/oem/11_persistency.yaml

```diff
@@ -36,15 +36,21 @@ stages:
           /usr/share/pki/trust/anchors
           /var/lib/ca-certificates
         PERSISTENT_STATE_BIND: "true"
-    - if: |
-        [ -r /run/cos/custom-layout.env ]
+    - if: '[ -f "/run/cos/recovery_mode" ]'
+      name: "Layout configuration for recovery"
+      environment_file: /run/cos/cos-layout.env
+      # We omit COS_PERSISTENT here and have no RW_PATHS, or BINDS
+      # RW_PATHS will be set to the default, which works for recovery
+      environment:
+        VOLUMES: "LABEL=COS_OEM:/oem"
+        OVERLAY: "tmpfs:25%"
+    - if: '[ -r /run/cos/custom-layout.env ] && [ ! -f "/run/cos/recovery_mode" ] && [ ! -f /run/cos/live_mode ]'
       name: "append custom bind and ephemeral mounts to /run/cos/cos-layout.env"
       commands:
         - |
              source /run/cos/cos-layout.env
              source /run/cos/custom-layout.env
              PERSISTENT_STATE_PATHS="${CUSTOM_EPHEMERAL_MOUNTS} ${PERSISTENT_STATE_PATHS}"
-             RW_PATHS="${CUSTOM_BIND_MOUNTS} ${RW_PATHS}"
```

Comment:
is this going to be read by immucore? the custom mount is injecting RW_PATHS in this case to provide extra ephemeral paths

Reply:
this is wrong, it needs to inject the ephemeral paths but its doing bind mounts! There is another patch incoming that reworks this, currently testing it locally

Reply:
ouch!